### PR TITLE
allows for plus in username, addresses issue #10414

### LINKED ISSUE #10414: [Issue]: Jellyfin does not allow special characters such as '+' in user names
### Please describe your bug

When creating a new user via API or Web interface, if a user's name has a special character such as commonly used '+' in email addresses, 400 error is returned.

### Jellyfin Version

10.8.z

### if other:

_No response_

### Environment

```markdown
- OS:
- Linux Kernel:
- Virtualization:
- Clients:
- Browser:
- FFmpeg Version:
- Playback Method:
- Hardware Acceleration:
- GPU Model:
- Plugins:
- Reverse Proxy:
- Base URL:
- Networking:
- Storage:
```

### Jellyfin logs

_No response_

### FFmpeg logs

_No response_

### Please attach any browser or client logs here

_No response_

### Please attach any screenshots here

_No response_

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by BenSanex**
**Changes**
Updates regex to include "+" in allowed characters, adds more test cases with accented characters.

**Issues**
Fixes #10414

### CODE REVIEW

File: Jellyfin.Server.Implementations/Users/UserManager.cs [hunk 1]
```diff
@@ -103,7 +103,7 @@ public partial class UserManager : IUserManager
         // This is some regex that matches only on unicode "word" characters, as well as -, _ and @
```
Comment (barronpm) [inline Jellyfin.Server.Implementations/Users/UserManager.cs:103]:
```suggestion
        // This is some regex that matches only on unicode "word" characters, as well as -, _, @, and +
```

**Comment by BenSanex** [comment]
Sorry I was slow on getting that suggestion in

**Review by 1337joe — APPROVED** [review]

File: Jellyfin.Server.Implementations/Users/UserManager.cs [hunk 2]
```diff
@@ -105,10 +105,10 @@ public partial class UserManager : IUserManager
         /// <inheritdoc/>
         public IEnumerable<Guid> UsersIds => _users.Keys;
 
-        // This is some regex that matches only on unicode "word" characters, as well as -, _ and @
+        // This is some regex that matches only on unicode "word" characters, as well as -, _, @, and +
         // In theory this will cut out most if not all 'control' characters which should help minimize any weirdness
-        // Usernames can contain letters (a-z + whatever else unicode is cool with), numbers (0-9), at-signs (@), dashes (-), underscores (_), apostrophes ('), periods (.) and spaces ( )
-        [GeneratedRegex(@"^[\w\ \-'._@]+$")]
+        // Usernames can contain letters (a-z + whatever else unicode is cool with), numbers (0-9), at-signs (@), dashes (-), underscores (_), apostrophes ('), periods (.), plus-signs (+), and spaces ( )
+        [GeneratedRegex("^[\\w\\ \\-\\+'._@]+$")]
```
Comment (cvium) [inline Jellyfin.Server.Implementations/Users/UserManager.cs:111]: Why remove the `@`?

**Comment by JPVenson** [comment]
Does this needs a corresponding PR to change client side validation?

**Comment by Bond-009** [comment]
Superseded by #12819

**Pull request closed without merging**